# Issue 32839: Show underlying error for failed coercions

archive/issues_032602.json:
```json
{
    "assignees": [
        "https://github.com/Freax13"
    ],
    "body": "<div id=\"comment:0\"></div>\n\nAbout 14 years ago, someone left a comment that we should really report the underlying error for failed coercions. This ticket aims to do just that by using python's exception chaining feature.\n\nComponent: **coercion**\n\nKeywords: **error reporting**\n\nBranch/Commit: **[u/gh-Freax13/show_underlying_error_for_failed_coercions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-Freax13/show_underlying_error_for_failed_coercions) @ [`c3289ba`](https://github.com/sagemath/sagetrac-mirror/commit/c3289ba3d85a47390a031e0f40d8600ced1b3291)**\n\n_Issue created by migration from https://trac.sagemath.org/ticket/32839_\n\n",
    "created_at": "2021-11-08T22:04:20Z",
    "labels": [
        "https://github.com/sagemath/sage/labels/c%3A%20coercion",
        "https://github.com/sagemath/sage/labels/p%3A%20minor%20/%204",
        "https://github.com/sagemath/sage/labels/enhancement"
    ],
    "reactions": [],
    "repository": "https://github.com/sagemath/sage",
    "title": "Show underlying error for failed coercions",
    "type": "issue",
    "updated_at": "2023-01-29T23:06:23Z",
    "url": "https://github.com/sagemath/sage/issues/32839",
    "user": "https://github.com/Freax13"
}
```
<div id="comment:0"></div>

About 14 years ago, someone left a comment that we should really report the underlying error for failed coercions. This ticket aims to do just that by using python's exception chaining feature.

Component: **coercion**

Keywords: **error reporting**

Branch/Commit: **[u/gh-Freax13/show_underlying_error_for_failed_coercions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-Freax13/show_underlying_error_for_failed_coercions) @ [`c3289ba`](https://github.com/sagemath/sagetrac-mirror/commit/c3289ba3d85a47390a031e0f40d8600ced1b3291)**

_Issue created by migration from https://trac.sagemath.org/ticket/32839_





---

archive/issue_events_449815.json:
```json
{
    "actor": "https://github.com/Freax13",
    "created_at": "2021-11-08T22:04:20Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.5",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449815"
}
```



---

archive/issue_events_449816.json:
```json
{
    "actor": "https://github.com/Freax13",
    "created_at": "2021-11-08T22:04:20Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20major%20/%203",
    "label_color": "ffbb00",
    "label_name": "p: major / 3",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449816"
}
```



---

archive/issue_comments_530179.json:
```json
{
    "body": "Branch: **[u/gh-Freax13/show_underlying_error_for_failed_coercions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-Freax13/show_underlying_error_for_failed_coercions)**",
    "created_at": "2021-11-08T22:06:31Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32839#issuecomment-530179",
    "user": "https://github.com/Freax13"
}
```

Branch: **[u/gh-Freax13/show_underlying_error_for_failed_coercions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-Freax13/show_underlying_error_for_failed_coercions)**



---

archive/issue_comments_530180.json:
```json
{
    "body": "Description changed:\n``````diff\n--- \n+++ \n@@ -1 +1 @@\n-\n+About 14 years ago, someone left a comment that we should really report the underlying error for failed coercions. This ticket aims to do just that by using python's exception chaining feature.\n``````\n",
    "created_at": "2021-11-08T22:10:41Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32839#issuecomment-530180",
    "user": "https://github.com/Freax13"
}
```

Description changed:
``````diff
--- 
+++ 
@@ -1 +1 @@
-
+About 14 years ago, someone left a comment that we should really report the underlying error for failed coercions. This ticket aims to do just that by using python's exception chaining feature.
``````




---

archive/issue_comments_530181.json:
```json
{
    "body": "<div id=\"comment:2\"></div>\n\nNew commits:\n<table><tr><td><a href=\"https://github.com/sagemath/sagetrac-mirror/commit/c3289ba3d85a47390a031e0f40d8600ced1b3291\"><code>c3289ba</code></a></td><td><code>report the underlying error for failed coercions</code></td></tr></table>\n",
    "created_at": "2021-11-08T22:10:41Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32839#issuecomment-530181",
    "user": "https://github.com/Freax13"
}
```

<div id="comment:2"></div>

New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c3289ba3d85a47390a031e0f40d8600ced1b3291"><code>c3289ba</code></a></td><td><code>report the underlying error for failed coercions</code></td></tr></table>




---

archive/issue_events_449817.json:
```json
{
    "actor": "https://github.com/Freax13",
    "created_at": "2021-11-08T22:10:41Z",
    "event": "assigned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "subject": "https://github.com/Freax13",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449817"
}
```



---

archive/issue_events_449818.json:
```json
{
    "actor": "https://github.com/Freax13",
    "created_at": "2021-11-08T22:10:41Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "label": "https://github.com/sagemath/sage/labels/enhancement",
    "label_color": "696969",
    "label_name": "enhancement",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449818"
}
```



---

archive/issue_events_449819.json:
```json
{
    "actor": "https://github.com/Freax13",
    "created_at": "2021-11-08T22:10:41Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "label": "https://github.com/sagemath/sage/labels/c%3A%20coercion",
    "label_color": "0000ff",
    "label_name": "c: coercion",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449819"
}
```



---

archive/issue_comments_530182.json:
```json
{
    "body": "Commit: **[`c3289ba`](https://github.com/sagemath/sagetrac-mirror/commit/c3289ba3d85a47390a031e0f40d8600ced1b3291)**",
    "created_at": "2021-11-08T22:10:41Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32839#issuecomment-530182",
    "user": "https://github.com/Freax13"
}
```

Commit: **[`c3289ba`](https://github.com/sagemath/sagetrac-mirror/commit/c3289ba3d85a47390a031e0f40d8600ced1b3291)**



---

archive/issue_comments_530183.json:
```json
{
    "body": "Changed keywords from none to **error reporting**",
    "created_at": "2021-11-08T22:10:41Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/32839#issuecomment-530183",
    "user": "https://github.com/Freax13"
}
```

Changed keywords from none to **error reporting**



---

archive/issue_events_449820.json:
```json
{
    "actor": "https://github.com/Freax13",
    "created_at": "2021-11-08T22:10:41Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20major%20/%203",
    "label_color": "ffbb00",
    "label_name": "p: major / 3",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449820"
}
```



---

archive/issue_events_449821.json:
```json
{
    "actor": "https://github.com/Freax13",
    "created_at": "2021-11-08T22:10:41Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20minor%20/%204",
    "label_color": "ffe799",
    "label_name": "p: minor / 4",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449821"
}
```



---

archive/issue_events_449822.json:
```json
{
    "actor": "https://github.com/mkoeppe",
    "created_at": "2021-12-18T19:11:26Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.5",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449822"
}
```



---

archive/issue_events_449823.json:
```json
{
    "actor": "https://github.com/mkoeppe",
    "created_at": "2021-12-18T19:11:26Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.6",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449823"
}
```



---

archive/issue_events_449824.json:
```json
{
    "actor": "https://github.com/mkoeppe",
    "created_at": "2022-04-01T21:16:35Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.6",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449824"
}
```



---

archive/issue_events_449825.json:
```json
{
    "actor": "https://github.com/mkoeppe",
    "created_at": "2022-04-01T21:16:35Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.7",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449825"
}
```



---

archive/issue_events_449826.json:
```json
{
    "actor": "https://github.com/mkoeppe",
    "created_at": "2022-08-31T02:51:13Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.7",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449826"
}
```



---

archive/issue_events_449827.json:
```json
{
    "actor": "https://github.com/mkoeppe",
    "created_at": "2022-08-31T02:51:13Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.8",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449827"
}
```



---

archive/issue_events_449828.json:
```json
{
    "actor": "https://github.com/mkoeppe",
    "created_at": "2023-01-29T23:06:23Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/32839",
    "milestone_number": null,
    "milestone_title": "sage-9.8",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/32839#event-449828"
}
```
